cmd/labs/project/entrypoint.go

```diff
@@ -0,0 +1,247 @@
+		e.Installer.MinRuntimeVersion = "13.1"
+	}
+	return e.Installer.MinRuntimeVersion != ""
+}
+
+func (e *Entrypoint) NeedsWarehouse() bool {
+	if e.Installer == nil {
+		return false
+	}
+	return len(e.Installer.WarehouseTypes) != 0
+}
+
+func (e *Entrypoint) Prepare(cmd *cobra.Command) (map[string]string, error) {
+	ctx := cmd.Context()
+	libDir := e.EffectiveLibDir(ctx)
+	environment := map[string]string{
+		"DATABRICKS_CLI_VERSION":     build.GetInfo().Version,
+		"DATABRICKS_LABS_CACHE_DIR":  e.CacheDir(ctx),
+		"DATABRICKS_LABS_CONFIG_DIR": e.ConfigDir(ctx),
+		"DATABRICKS_LABS_STATE_DIR":  e.StateDir(ctx),
+		"DATABRICKS_LABS_LIB_DIR":    libDir,
+	}
+	if e.IsPythonProject(ctx) {
+		e.preparePython(ctx, environment)
+	}
+	cfg, err := e.validLogin(cmd)
+	if err != nil {
+		return nil, fmt.Errorf("login: %w", err)
+	}
+	// cleanup auth profile and config file location,
+	// so that we don't confuse SDKs
+	cfg.Profile = ""
+	cfg.ConfigFile = ""
+	varNames := []string{}
+	for k, v := range e.environmentFromConfig(cfg) {
+		environment[k] = v
+		varNames = append(varNames, k)
+	}
+	if e.NeedsCluster() && e.RequireRunningCluster {
+		err = e.ensureRunningCluster(ctx, cfg)
+		if err != nil {
+			return nil, fmt.Errorf("running cluster: %w", err)
+		}
+	}
+	log.Debugf(ctx, "Passing down environment variables: %s", strings.Join(varNames, ", "))
+	return environment, nil
+}
+
+func (e *Entrypoint) preparePython(ctx context.Context, environment map[string]string) {
+	venv := e.virtualEnvPath(ctx)
+	environment["PATH"] = e.joinPaths(filepath.Join(venv, "bin"), env.Get(ctx, "PATH"))
+
+	// PYTHONPATH extends the standard lookup locations for module files. It follows the same structure as
+	// the shell's PATH, where you specify one or more directory paths separated by the appropriate delimiter
+	// (such as colons for Unix or semicolons for Windows). If a directory listed in PYTHONPATH doesn't exist,
+	// it is disregarded without any notifications.
+	//
+	// Beyond regular directories, individual entries in PYTHONPATH can point to zipfiles that contain pure
+	// Python modules in either their source or compiled forms. It's important to note that extension modules
+	// cannot be imported from zipfiles.
+	//
+	// The initial search path varies depending on your installation but typically commences with the
+	// prefix/lib/pythonversion path (as indicated by PYTHONHOME). This default path is always included
+	// in PYTHONPATH.
+	//
+	// An extra directory can be included at the beginning of the search path, coming before PYTHONPATH,
+	// as explained in the Interface options section. You can control the search path from within a Python
+	// script using the sys.path variable.
+	//
+	// Here we are also supporting the "src" layout for python projects.
+	//
+	// See https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH
+	libDir := e.EffectiveLibDir(ctx)
+	environment["PYTHONPATH"] = e.joinPaths(libDir, filepath.Join(libDir, "src"))
```

Comment:
What if PYTHONPATH is already set by the user? Should that be additive?

Reply:
The intention for every install is to be sandboxed - not dependent on anything else than Python binary. Having ability to override PYTHONPATH in the mix will break this assumption. Need strong evidence that this is really needed.